bytestring/common/src/ByteString.kt

```diff
@@ -0,0 +1,467 @@
+/*
+ * Copyright 2010-2023 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+ */
+
+/*
+ * Copyright (C) 2018 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kotlinx.io.bytestring
+
+import kotlin.math.max
+import kotlin.math.min
+
+private val HEX_DIGITS = "0123456789ABCDEF".toCharArray()
+
+/**
+ * Wraps given [bytes] into a byte string.
+ *
+ * @param bytes a sequence of bytes to be wrapped.
+ */
+public fun ByteString(vararg bytes: Byte): ByteString = if (bytes.isEmpty()) {
+    ByteString.EMPTY
+} else {
+    ByteString.wrap(bytes)
+}
+
+/**
+ * An immutable wrapper around a byte sequence providing [String] like functionality.
```

Comment:
It would be nice to document it extensively (how it can be used, what for and what it is semantically) in the follow-up PR

Reply:
https://github.com/Kotlin/kotlinx-io/pull/158